### ISSUE TITLE
Bug in OpaLexer with unquoted attributes

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#548]
(Original issue 841 created by skleinfeld on 2013-02-01T22:32:10.452836+00:00)

The following code listing is highlighted properly by the OpaLexer:

````
function sample_page() {
  <header>
    <h3>HTML in Opa</h3>
  </header>
  <article>
    <div class="container">
      <p>Learning by examples.</p>
    </div>
  </article>
}

````

But the following valid Opa syntax (with an unquoted attribute on the div) is not highlighted properly:

````
function sample_page() {
  <header>
    <h3>HTML in Opa</h3>
  </header>
  <article>
    <div class=container>
      <p>Learning by examples.</p>
    </div>
  </article>
}
````

And throws the following error:

````
IndexError: list index out of range
   (file "/usr/local/lib/python2.6/dist-packages/Pygments-1.5-py2.6.egg/pygments/lexer.py", line 541, in get_tokens_unprocessed)
````

It looks like this is an issue stemming from the OpaLexer's 'html-attr-value' section, which has no regex handling for unquoted attributes:

````

        'html-attr-value': [
            (r"'", String.Single, ('#pop', 'single-string')),
            (r'"', String.Single, ('#pop', 'string')),
            (r'#'+ident_re, String.Single, '#pop'),
            (r'#(?={)', String.Single, ('#pop', 'root')),
            (r'{', Operator, ('#pop', 'root')), # this is a tail call!                                                                           
        ],
````

Thanks,
Sanders

---

Sanders Kleinfeld

Publishing Technology Engineer

O'Reilly Media Inc.

sanders@oreilly.com

@sandersk

**Comment by Anteru** [comment]
(Original issue 841 last updated on 2013-02-03T09:23:39.691172+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2013-02-03T09:23:39.685889+00:00)

Closes #841: fix handling for unquoted attribute values in Opa.

→ <<cset 41605f24745e>>

**Comment by Anteru** [comment]
(Original comment by skleinfeld on 2013-02-05T03:35:05.787796+00:00)

(Reply via san...@oreilly.com):

Wow, thank you very much for the quick response and fix!

Best,
Sanders
